### PR TITLE
update release steps for gsg

### PR DESCRIPTION
**Pull request by garyverhaegen-da**
Add steps to ensure we end up with a working published GSG for stable
releases.

CHANGELOG_BEGIN
CHANGELOG_END

### CODE REVIEW

File: release/RELEASE.md [hunk 1]
```diff
@@ -198,6 +204,10 @@ latest commit on master.
    ./release.sh changes stable previous
    ```
 
+1. **[STABLE]** Update the
+   [create-daml-app](https://github.com/digital-asset/create-daml-app) template
+   to use the new version, and add a tag for it.
```
Comment (rohanjr) [inline release/RELEASE.md]: We should say that the tag needs to be `vX.XX.XX` (the new version number).

Comment (rohanjr) [reply release/RELEASE.md]: If the previous step (create-daml-app being on the latest snapshot) was done, then this should just be a matter of changing version numbers, though in several places (daml.yaml and package.json at least). Just confirming we want this burden on the releaser @garyverhaegen-da @hurryabit ?

File: release/RELEASE.md [hunk 2]
```diff
@@ -25,6 +25,12 @@ latest commit on master.
    ./release.sh changes stable latest
    ```
 
+1. **[STABLE]** Check that the [Getting Started
+   Guide](https://github.com/digital-asset/daml/tree/master/docs/source/getting-started)
+   in the snapshot you have chosen to promote to a stable release is supported
+   by the latest
+   [create-daml-app](https://github.com/digital-asset/create-daml-app).
```
[the remark this answers is not in the document]
Comment (hurryabit) [reply release/RELEASE.md]: We don't want `create-daml-app` to be on the latest snapshot but on the one we intend to promote to a stable release. Other than that it sounds good to me.

File: release/RELEASE.md (hunk 1 above)
Comment (hurryabit) [reply release/RELEASE.md]: I think to put that burden on the person promoting the release is fine for now. We plan to do this once a month and the person is most likely somebody who'll specifically be asked to make the release, including all such hassles.

**Comment by garyverhaegen-da** [comment]
I think this will be superseded by #5259. Once create-daml-app is in this repo there should not be much risk of desync and therefore no need to pay particular attention to it during release.

**Pull request closed without merging**